src/socket-to-conn.ts

```diff
@@ -38,7 +38,17 @@ export const toMultiaddrConnection = (socket: Socket, options?: ToConnectionOpti
     options.localAddr = options.remoteAddr
   }
 
-  const remoteAddr = options.remoteAddr ?? toMultiaddr(socket.remoteAddress ?? '', socket.remotePort ?? '')
+  let remoteAddr: Multiaddr
+
+  if (options.remoteAddr != null) {
+    remoteAddr = options.remoteAddr
+  } else {
+    const address = socket.address()
+
+    // @ts-expect-error type of address is `{} | AdressInfo` - how to exclude {}?
```

Comment:
I think you can explicitly type `address` above.
`const address: AddressInfo = ...`